Avoid network errors when a large number of files are served with http2 ERR_HTTP2_PROTOCOL_ERROR

GabrielDelepine opened the pull request:
## Issue

I am experiencing network errors. It is due to the large number of files my page is loading (300 to 400 hundred).

The number of files shouldn't be a problem, especially in a project that advocate ESM, importmap, modulepreload, http2/3 multiplexing, etc...

![ERR_HTTP2_PROTOCOL_ERROR](https://github.com/modernweb-dev/web/assets/1979087/d627b200-2dbd-46ee-a1d3-7cb73df994a7)

## Solution

Raise-up the maxSessionMemory of the http2 server from the default value (10) to 20

koddsson commented:
Hey @GabrielDelepine; this looks ok but I'm slightly vary of the jump from the default 10 megabytes to 300 megabytes. Can you add a changeset to this as well?

koddsson commented:
I also want tot land #2467 before this one so we can at least verify that these changes pass the "smoke" test.

GabrielDelepine commented:
> Hey @GabrielDelepine; this looks ok but I'm slightly vary of the jump from the default 10 megabytes to 300 megabytes. Can you add a changeset to this as well?

Good point, the default value is `10`, `300` is too rough.
I am currently performing some tests with more reasonable values

koddsson commented:
> I also want to land #2467 before this one so we can at least verify that these changes pass the "smoke" test.

I've landed this so if you rebase this PR on `main` we'll have the new `http2` test for a little bit of more confidence.

GabrielDelepine commented:
Thanks, the branch is now rebased with `master`.

After conducting some performance tests, a value of 20 is enough to serve few hundred files per page. I updated the PR accordingly
